src/lib.rs

```diff
@@ -496,28 +506,29 @@ pub fn run(args: &Arguments, mut tests: Vec<Trial>) -> Conclusion {
         }
     } else {
         // Run test in thread pool.
-        let pool = match args.test_threads {
-            Some(num_threads) => ThreadPool::new(num_threads),
-            None => ThreadPool::default()
-        };
+        let num_tests = tests.len();
         let (sender, receiver) = mpsc::channel();
 
-        let num_tests = tests.len();
-        for test in tests {
+        let mut tasks: Vec<pool::BoxedTask> = Default::default();
```

Comment:
Shouldn't this work?

```suggestion
        let mut tasks = Vec::new();
```